packages/backend/src/open_payments/quote/service.ts

```diff
@@ -195,55 +192,87 @@ export async function resolveReceiver(
   return receiver
 }
 
-interface CalculateQuoteAmountsAfterFeesResult {
+interface CalculateQuoteAmountsWithFeesResult {
   receiveAmountValue: bigint
   debitAmountValue: bigint
 }
 
-function calculateQuoteAmountsAfterFees(
+/**
+ * Calculate fixed-send quote amounts: debitAmount is locked,
+ * subtract fees (considering the exchange rate) from the receiveAmount.
+ */
+function calculateFixedSendQuoteAmounts(
   deps: ServiceDependencies,
   quote: Quote,
-  maxReceiveAmountValue?: bigint
-): CalculateQuoteAmountsAfterFeesResult {
-  let debitAmountValue = quote.debitAmount.value
-  let receiveAmountValue = quote.receiveAmount.value
-
-  if (!maxReceiveAmountValue) {
-    // FixedDelivery
-    const fees = quote.fee?.calculate(debitAmountValue) ?? 0n
-    debitAmountValue = BigInt(debitAmountValue) + fees
-    if (
-      debitAmountValue < quote.debitAmount.value ||
-      debitAmountValue <= BigInt(0)
-    ) {
-      throw QuoteError.InvalidAmount
-    }
-  } else {
```

Comment:
split these up into functions, logic remains the same other than the comment below